client/components/button-group/test/index.js

```diff
@@ -1,35 +1,30 @@
 /**
  * External dependencies
  */
-import { assert } from 'chai';
 import { shallow } from 'enzyme';
 import React from 'react';
-import sinon from 'sinon';
 
 /**
  * Internal dependencies
  */
-
 import { Button } from '@automattic/components';
 
 describe( 'ButtonGroup', () => {
-	let sandbox, ButtonGroup;
+	let ButtonGroup, consoleErrorSpy;
 
-	beforeEach( () => {
-		sandbox = sinon.createSandbox();
-		sandbox.stub( console, 'error' );
-		sandbox.stub( console, 'log' );
+	beforeAll( async () => {
+		consoleErrorSpy = jest.spyOn( global.console, 'error' ).mockImplementation();
```

Comment:
`beforeAll` and `afterAll` run only once when the parent `describe()` block starts and ends. Therefore, the last test ("should throw an error ...") is checking a console mock that could have been called by one of previous tests.

Why didn't you leave the original `beforeEach` and `afterEach`? That recreates a fresh mock for each of the tests.

Also, the mock seems to be needed only by one test. It doesn't need to be created for the other ones.

Reply:
> `beforeAll` and `afterAll` run only once when the parent `describe()` block starts and ends. Therefore, the last test ("should throw an error ...") is checking a console mock that could have been called by one of previous tests.
> 
> Why didn't you leave the original `beforeEach` and `afterEach`? That recreates a fresh mock for each of the tests.

My bad, didn't think that through :man_facepalming: 

> Also, the mock seems to be needed only by one test. It doesn't need to be created for the other ones.

Good point, thanks! I'll fix :+1:

Reply:
a76f94f989